File: sdk/src/main/java/org/coralibre/android/sdk/internal/datatypes/DiagnosisKey.java
package org.coralibre.android.sdk.internal.datatypes;

import org.coralibre.android.sdk.fakegms.nearby.exposurenotification.ExposureConfiguration;

public class DiagnosisKey {

    private final TemporaryExposureKey_internal tek;

    /**
     * A number in range [0, 7]
     * @see ExposureConfiguration#getTransmissionRiskScore(int)
     */
    private final int transmissionRiskLevel;

    public DiagnosisKey(final TemporaryExposureKey_internal tek,
                        final int transmissionRiskLevel) {
        this.tek = tek;
        this.transmissionRiskLevel = transmissionRiskLevel;
    }

    public byte[] getKeyData() {
        return tek.getKey();
    }

    public ENInterval getInterval() {
        return tek.getInterval();
    }

    public int getTransmissionRiskLevel() {
        return transmissionRiskLevel;
    }
}

File: sdk/src/main/java/org/coralibre/android/sdk/internal/datatypes/DiagnosisKey.kt
package org.coralibre.android.sdk.internal.datatypes

import org.coralibre.android.sdk.fakegms.nearby.exposurenotification.ExposureConfiguration

class DiagnosisKey(
    private val tek: InternalTemporaryExposureKey,
    /**
     * A number in range [0, 7]
     * @see ExposureConfiguration.getTransmissionRiskScore
     */
    val transmissionRiskLevel: Int
) {
    val keyData: ByteArray
        get() = tek.key
    val interval: ENInterval
        get() = tek.interval
}
